oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/http/OAuth2ErrorResponseErrorHandler.java

```diff
@@ -70,7 +70,7 @@ private OAuth2Error readErrorFromWwwAuthenticate(HttpHeaders headers) {
 			return null;
 		}
 		BearerTokenError bearerTokenError = getBearerToken(wwwAuthenticateHeader);
-		String errorCode = (bearerTokenError.getCode() != null) ? bearerTokenError.getCode()
+		String errorCode = (bearerTokenError != null && bearerTokenError.getCode() != null) ? bearerTokenError.getCode()
```

Comment:
This doesn't appear to address the issue completely since `bearerTokenError` is referred to later on in the method, potentially causing more NPEs.

If wonder if it would be better to do:

```java
if (bearerTokenError == null) {
    return new OAuth2Error(OAuth2ErrorCodes.SERVER_ERROR, null, null);
}
String errorCode = ...
```

Reply:
That's a good point. I will do that.